gui.py

```diff
@@ -144,36 +215,109 @@ def callback_run_script(self):
+            self.after(100, self.listen_for_result)
         else:
             # If exception was raised print to the log
             if exc is not None:
                 self.log.add_message(
                         "\n".join(traceback.format_exception_only(exc[0], exc[1])),
                         color="RED")
                 traceback.print_tb(exc[2])
-            self.b["state"] = "normal"
-        #self.new_thread.join(0.1)
+            toggle_widgets(self.main_frame, "normal")
+            self.progress.stop()
+            self.new_thread = None
+        
+        
+        
+    def export_settings(self):
+        """
+        Export the current settings to a log file in the corresponding "Runs"
+        directory.
+        """
+        args = {self.user_names[x]:self.vars[x].get() for x in self.user_names}
+        factor_files = {"RTF File": self.rtf_file.get(),
+                        "PTF File": self.ptf_file.get(),
+                        "Airport Growth File": self.airport_growth.get()}
+        args = {**args, **factor_files}
+        output_dir = os.path.join(self.vars["tmfs_root"].get(), "Runs", 
+                                  self.vars["tel_year"].get(), "Demand", 
+                                  self.vars["tel_id"].get())
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+            
+            self.log.add_message("Created New Directory - {}".format(
+                    output_dir))
+            
+        file_path = os.path.join(output_dir, "settings.txt")
+        with open(file_path, "w", newline="") as f:
+            w = csv.writer(f)
+            w.writerows(args.items())
+            
+        self.log.add_message("Exported Settings to {}".format(file_path))
+    
+    
+    def import_settings(self):
+        """
+        Load a previously exported settings file. Does not check content of
+        settings file.
+        """
+        file_path = filedialog.askopenfilename(
+                parent=self.main_frame, title="Select Settings File",
+                filetypes=[("text files", "*.txt")])
+        if file_path == "":
+            return
+        
+        reverse_user_names = {v:k for k, v in self.user_names.items()}
+        factor_files = {}
+        with open(file_path, "r") as f:
+            r = csv.reader(f)
+            for row in r:
+                try:
+                    var_name = reverse_user_names[row[0]]
+                    self.vars[var_name].set(row[1])
+                except KeyError:
+                    factor_files[row[0]] = row[1]
+        self.on_directory_update(self.vars["tmfs_root"].get())
+        # If there are settings available for the factors files then set them
+        try:
+            self.rtf_file.set(factor_files["RTF File"])
+            self.ptf_file.set(factor_files["PTF File"])
+            self.airport_growth.set(factor_files["Airport Growth File"])
+        except:
+            pass

```

Comment:
I've amended this so that it uses `asksaveasfilename` and a json file instead. The rtf/pt/airport factor files are also now in the same format as the other parameters so it is a bit tidier.